WooCommerce/Classes/ViewRelated/Dashboard/Settings/Help/SupportForm/SupportForm.swift

```diff
@@ -7,13 +7,94 @@ final class SupportFormHostingController: UIHostingController<SupportForm> {
 
     init(viewModel: SupportFormViewModel) {
         super.init(rootView: SupportForm(viewModel: viewModel))
+        handleSupportRequestCompletion(viewModel: viewModel)
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        createZendeskIdentity()
+    }
+
+    /// Creates the Zendesk Identity if needed.
+    /// If it fails, it pops back the view and informs the user.
+    ///
+    func createZendeskIdentity() {
+        // TODO: We should consider refactoring this to present the email alert using SwiftUI.
+        ZendeskProvider.shared.createIdentity(presentIn: self) { [weak self] identityCreated in
+            if !identityCreated {
+                self?.logIdentityErrorAndPopBack()
+            }
+        }
+    }
+
+    /// Registers a completion block on the view model to properly show alerts and notices.
+    ///
+    func handleSupportRequestCompletion(viewModel: SupportFormViewModel) {
+        viewModel.onCompletion = { [weak self] result in
+            guard let self else { return }
+            switch result {
+            case .success:
+                self.informSuccessAndPopBack()
+            case .failure(let error):
+                self.logAndInformErrorCreatingRequest(error)
+            }
+        }
     }
 
     required dynamic init?(coder aDecoder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
 }
 
+private extension SupportFormHostingController {
+    /// Shows an alert informing the support creation success and after confirmation pops the view back.
+    ///
+    func informSuccessAndPopBack() {
+        let alertController = UIAlertController(title: Localization.requestSent,
+                                                message: Localization.requestSentMessage,
+                                                preferredStyle: .alert)
+        alertController.addDefaultActionWithTitle(Localization.gotIt) { _ in
+            self.navigationController?.popViewController(animated: true)
+        }
+        present(alertController, animated: true)
+    }
+
+    /// Logs and informs the user that a support request could not be created
+    ///
+    func logAndInformErrorCreatingRequest(_ error: Error) {
+        let notice = Notice(title: Localization.requestSentError, feedbackType: .error)
+        let noticePresenter = DefaultNoticePresenter()
+        noticePresenter.presentingViewController = self
+        noticePresenter.enqueue(notice: notice)
+
+        DDLogError("⛔️ Could not create Support Request. Error: \(error.localizedDescription)")
+    }
+
+    /// Informs user about identity error and pop back
+    ///
+    func logIdentityErrorAndPopBack() {
+        let notice = Notice(title: Localization.badIdentityError, feedbackType: .error)
+        ServiceLocator.noticePresenter.enqueue(notice: notice)
```

Comment:
The notice is being enqueued but I don't see it presented on screen when I tap "Cancel":

https://user-images.githubusercontent.com/8658164/219409722-422b9db4-f37d-4ea0-a8ef-71de70cd255f.mp4

Note: I tested this after tapping "Help" before I finished logging in, and I see the screen recording you shared is in the logged-in context.

Reply:
Nice catch, This happens because the `ServiceLocator.noticePresenter` does not work well when there is a modal on top. I've fixed it by using a local notice presenter here https://github.com/woocommerce/woocommerce-ios/pull/8865/commits/df1a1bb2a8145c77a1fd0075f1e1bacb23dea731

https://user-images.githubusercontent.com/562080/219490228-8a883dd8-c482-4303-80a8-ff4b652e9f42.mov

Reply:
Ah, that's right! I remembered that it didn't work well everywhere but couldn't remember exactly what the cause was. This fix works great!